org.sonarlint.eclipse.core/src/org/sonarlint/eclipse/core/internal/jobs/AnalyzeProjectJob.java

```diff
@@ -226,79 +245,101 @@ private void runAnalysisAndUpdateMarkers(Map<IFile, IDocument> docPerFiles, fina
     return pathMatchersForTests;
   }
 
-  private static Collection<ProjectConfigurator> configure(final IProject project, Collection<IFile> filesToAnalyze, Map<IFile, String> fileLanguages,
+  private static Collection<ProjectConfigurator> configureDeprecated(final ISonarLintProject project, Collection<ISonarLintFile> filesToAnalyze,
     final Map<String, String> extraProperties,
     final IProgressMonitor monitor) {
-    ProjectConfigurationRequest configuratorRequest = new ProjectConfigurationRequest(project, filesToAnalyze, fileLanguages, extraProperties);
-    Collection<ProjectConfigurator> configurators = ConfiguratorUtils.getConfigurators();
     Collection<ProjectConfigurator> usedConfigurators = new ArrayList<>();
-    for (ProjectConfigurator configurator : configurators) {
+    if (project.getUnderlyingProject() != null) {
+      ProjectConfigurationRequest configuratorRequest = new ProjectConfigurationRequest(project.getUnderlyingProject(),
+        filesToAnalyze.stream()
+          .map(ISonarLintFile::getUnderlyingFile)
+          .filter(Objects::nonNull)
+          .collect(Collectors.toList()),
+        extraProperties);
+      Collection<ProjectConfigurator> configurators = SonarLintCorePlugin.getExtensionTracker().getConfigurators();
```

Comment:
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png 'Severity: MINOR') Remove this use of "ProjectConfigurator"; it is deprecated. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3ACallToDeprecatedMethod)